### PR TITLE
Adding OpenShift/EKS instructions for WireGuard installation

### PR DESCRIPTION
**Pull request by realgaurav**
## Description

Adding OpenShift/EKS instructions for WireGuard installation

## Release Note

```release-note
None required
```

### CODE REVIEW

File: security/encrypt-cluster-pod-traffic.md [hunk 1]
```diff
@@ -36,7 +36,92 @@ This how-to guide uses the following {{site.prodname}} features:
 
 1. Install WireGuard on cluster nodes using {% include open-new-window.html text='instructions for your operating system' url='https://www.wireguard.com/install/' %}. Note that you may need to reboot your nodes after installing WireGuard to make the kernel modules available on your system.
 
-    > **Note**: Nodes that do not support WireGuard will not be secured by WireGuard tunnels, even if traffic running on the node to and from the pods goes to nodes that do support WireGuard.
+   Check the following if your operating system is not listed on the WireGuard installation page.
```
Comment (bmckercher123) [inline security/encrypt-cluster-pod-traffic.md:39]: Use the following instructions for operating systems not listed on the WireGuard installation page.

File: security/encrypt-cluster-pod-traffic.md [hunk 2]
```diff
@@ -36,7 +36,92 @@ This how-to guide uses the following {{site.prodname}} features:
...
+
+{% tabs %}
+<label:EKS,active:true>
+<%
+For installing WireGuard on default AMI image.
```
Comment (bmckercher123) [inline security/encrypt-cluster-pod-traffic.md:44]: To install WireGuard on the default AMI image:

File: security/encrypt-cluster-pod-traffic.md [hunk 3]
```diff
@@ -36,7 +36,92 @@ This how-to guide uses the following {{site.prodname}} features:
...
+   ```bash
+   sudo yum install kernel-devel-`uname -r` -y
+   sudo yum install https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm -y
+   sudo curl -o /etc/yum.repos.d/jdoss-wireguard-epel-7.repo https://copr.fedorainfracloud.org/coprs/jdoss/wireguard/repo/epel-7/jdoss-wireguard-epel-7.repo
+   sudo yum install wireguard-dkms wireguard-tools -y
+   ```
+%>
+<label:OpenShift>
+<%
+   For installing WireGuard for OpenShift v4.3:
```
Comment (bmckercher123) [inline security/encrypt-cluster-pod-traffic.md:55]: To install WireGuard for OpenShift v4.3:

File: security/encrypt-cluster-pod-traffic.md [hunk 4]
```diff
@@ -36,7 +36,92 @@ This how-to guide uses the following {{site.prodname}} features:
...
+   metadata:
+     labels:
+       machineconfiguration.openshift.io/role: worker
+     name: 10-kvc-wireguard-kmod
+   spec:
+     config:
+   EOF
+   ```
+
+   2. Create base Ignition config.
```
Comment (bmckercher123) [inline security/encrypt-cluster-pod-traffic.md:71]: Create the base Ignition config.

File: security/encrypt-cluster-pod-traffic.md [hunk 5]
```diff
@@ -36,7 +36,92 @@ This how-to guide uses the following {{site.prodname}} features:
...
+         "name": "require-kvc-wireguard-kmod.service",
+         "enabled": true,
+         "contents": "[Unit]\nRequires=kmods-via-containers@wireguard-kmod.service\n[Service]\nType=oneshot\nExecStart=/usr/bin/true\n\n[Install]\nWantedBy=multi-user.target"
+       }]
+     }
+   }
+   EOF
+   ```
+
+   3. Setup files
```
Comment (bmckercher123) [inline security/encrypt-cluster-pod-traffic.md:87]: Configure files.

File: security/encrypt-cluster-pod-traffic.md [hunk 6]
```diff
@@ -36,7 +36,92 @@ This how-to guide uses the following {{site.prodname}} features:
...
+   cd kmods-via-containers
+   make install DESTDIR=${FAKEROOT}/usr/local CONFDIR=${FAKEROOT}/etc/
+   cd ..
+   git clone https://github.com/realgaurav/kvc-wireguard-kmod
+   cd kvc-wireguard-kmod
+   make install DESTDIR=${FAKEROOT}/usr/local CONFDIR=${FAKEROOT}/etc/
+   cd ..
+   ```
+
+   4. Setup RPMs for kernel-core, kernel-devel and kernel-modules for the host kernel (can be found by running uname -r on the host). Update `$FAKEROOT/etc/kvc/wireguard-kmod.conf` for the RPM location.
```
Comment (bmckercher123) [inline security/encrypt-cluster-pod-traffic.md:100]: Configure RPMs...

File: security/encrypt-cluster-pod-traffic.md [hunk 7]
```diff
@@ -36,7 +36,92 @@ This how-to guide uses the following {{site.prodname}} features:
...
+   tar -x -C ${FAKEROOT} -f subs.tar.gz
+   ```
+
+   7. Get filetranspiler to generate the usable machine-config.
+   ```bash
+   git clone https://github.com/ashcrow/filetranspiler
+   ./filetranspiler/filetranspile -i ./wg-config.ign -f ${FAKEROOT} --format=yaml --dereference-symlinks | sed 's/^/     /' | (cat mc-wg-worker.yaml -) > mc-wg.yaml
+   ```
+
+   8. Finally with the KUBECONFIG set, run the following command to apply the MachineConfig created.
```
Comment (bmckercher123) [inline security/encrypt-cluster-pod-traffic.md:118]: With the KUBECONFIG set, run...

File: security/encrypt-cluster-pod-traffic.md [hunk 8]
```diff
@@ -36,7 +36,92 @@ This how-to guide uses the following {{site.prodname}} features:
 
 1. Install WireGuard on cluster nodes using {% include open-new-window.html text='instructions for your operating system' url='https://www.wireguard.com/install/' %}. Note that you may need to reboot your nodes after installing WireGuard to make the kernel modules available on your system.
```
Comment (bmckercher123) [inline security/encrypt-cluster-pod-traffic.md:37]: should we not just tell user to reboot after installing? Like this:

1. Install WireGuard on cluster nodes using <>
1. Reboot nodes.

File: security/encrypt-cluster-pod-traffic.md (hunk 4 above)
Comment (realgaurav) [reply security/encrypt-cluster-pod-traffic.md:70]: discussed offline, leaving as is.

File: security/encrypt-cluster-pod-traffic.md (hunk 8 above)
Comment (realgaurav) [reply security/encrypt-cluster-pod-traffic.md:37]: Reboot is not mandatory for all platforms e.g. EKS, Ubuntu in managed k8s case. It's however required for OpenShift.

File: security/encrypt-cluster-pod-traffic.md [hunk 9]
```diff
@@ -36,10 +36,94 @@ This how-to guide uses the following {{site.prodname}} features:
...
```
Comment (bmckercher123) [inline security/encrypt-cluster-pod-traffic.md:37]: To make this easier with numbering, I suggest making this just a paragraph.

Comment (realgaurav) [reply security/encrypt-cluster-pod-traffic.md:37]: I have removed the numbering in that section altogether. This also helps with numbered instructions in OpenShift mixing with the common instructions. PTAL.

File: security/encrypt-cluster-pod-traffic.md [hunk 10]
```diff
@@ -25,7 +25,7 @@ This how-to guide uses the following {{site.prodname}} features:
 > **Note**: WireGuard in {{site.prodname}} does not support IPv6 at this time. Also, encryption using WireGuard is not supported if `CALICO_NETWORKING_BACKEND=none` (e.g. managed Kubernetes platforms EKS, AKS and GKE).
 {: .alert .alert-info}
 
-### How to 
+### How to
 
 - [Enable WireGuard for a cluster](#enable-wireguard-for-a-cluster)
```
Comment (bmckercher123) [inline security/encrypt-cluster-pod-traffic.md:30]: - [Install WireGuard](#install-wireguard)

File: security/encrypt-cluster-pod-traffic.md [hunk 11]
```diff
@@ -34,23 +34,107 @@ This how-to guide uses the following {{site.prodname}} features:
...
+   ```
+
+   8. With with the KUBECONFIG set, run the following command to apply the MachineConfig created.
+   ```bash
+   oc create -f mc-wg.yaml
+   ```
+%>
+{% endtabs %}
+   > **Note**: Nodes that do not support WireGuard will not be secured by WireGuard tunnels, even if traffic running on the node to and from the pods goes to nodes that do support WireGuard.
    {: .alert .alert-info}
```
Comment (bmckercher123) [inline security/encrypt-cluster-pod-traffic.md:125]: #### Enable WireGuard

Use the following command to enable WireGuard encryption on all nodes.

**Review by bmckercher123 — APPROVED** [review]
approve

**Review by mgleung — APPROVED** [review]
LGTM

**Pull request merged**